src/array.cr

```diff
@@ -1545,11 +1544,46 @@ class Array(T)
     self
   end
 
+  # Returns `self` with all the elements shifted `n` times.
+  #
+  # ```
+  # a1 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  # a2 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Comment:
These parallel arrays make the sample difficult to grasp. I think its better to just use a single array (or at least only one literal). That perfectly demonstrates that each call to `#rotate!` builds on the result of the previous.